tests/unit/utils.py

```diff
@@ -226,6 +226,9 @@ def save_json_data(json_file: str, data: Dataset):
 
 def compare_result(actual: SingleDataset, expected: SingleDataset, rtol: float, atol: float | dict[str, float]):
     for key, expected_data in expected.items():
+        if not isinstance(expected_data, np.ndarray):
+            raise NotImplementedError()  # TODO(mgovers): Add support for columnar data
+
```

Comment:
shouldn't we fix this too?

Reply:
```suggestion
        if not isinstance(expected_data, np.ndarray):
            raise NotImplementedError("Validation is not implemented for columnar data")

```
Its not necessary to implement this. Validation tests would always convert to row based data.